firedrake/mesh.py

```diff
@@ -2479,20 +2512,60 @@ def _pic_swarm_in_mesh(parent_mesh, coords, fields=None, tolerance=None):
 
     coords, reference_coords, parent_cell_nums = \
         _parent_mesh_embedding(coords, parent_mesh, tolerance)
-    # mesh.topology.cell_closure[:, -1] maps Firedrake cell numbers to plex numbers.
-    plex_parent_cell_nums = parent_mesh.topology.cell_closure[parent_cell_nums, -1]
+
+    if parent_mesh.extruded:
+        # need to store the base parent cell number and the height to be able
+        # to map point coordinates back to the parent mesh
+        if parent_mesh.variable_layers:
+            # TODO: Don't know how variable layer cell numbering works - below
```

Comment:
I think the restriction here is that `locate_cell` doesn't know what to do. Determining the right base cell and height should still be relatively straightforward.

Reply:
Yes. We should probably just let `locate_cell` return `base_cell` and `height`; indeed these variables have once been computed in `locate_cell.c` (assuming uniform layer extrusion).
If `parent_mesh.variable_layers = True`, `locate_cell` should complain.

Reply:
Okay I'll get rid of the todo